### PR TITLE
Fix combined inv load

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
Racks now notify their parent on beeing loaded.
Building TE's no longer access unloaded areas for their combined storage

Fix traveler spawn half inside a wall

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/tileentities/TileEntityColonyBuilding.java [hunk 1]
```diff
@@ -454,16 +454,22 @@ public void markInvDirty()
...
-                    providers.removeIf(Objects::isNull);
-                    building.getAdditionalCountainers()
-                      .stream()
-                      .map(world::getTileEntity)
-                      .filter(entity -> entity instanceof TileEntityRack)
-                      .forEach(entity -> ((TileEntityRack) entity).setBuildingPos(this.pos));
+                    for (final BlockPos pos : building.getAdditionalCountainers())
+                    {
+                        if (WorldUtil.isBlockLoaded(world, pos))
+                        {
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/tileentities/TileEntityColonyBuilding.java:460]: Don't we need a "dirty" flag on load of the other invs too?

Comment (someaddons) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityColonyBuilding.java:460]: ye thats why I added to racks that they notify their building on load

Comment (Raycoms) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityColonyBuilding.java:460]: kk

**Comment by someaddons** [comment]
tested, works

**Pull request merged**